### PR TITLE
feat: Open MANA-FIAT Gateway Request Flow + No wallet address in wiget url

### PR DESCRIPTION
**Pull request by kevinszuchet**

### CODE REVIEW

File: src/modules/manaFiatGateway/sagas.ts [hunk 1]
```diff
@@ -56,7 +56,9 @@ function* handleOpenFiatGateway(
       transak.openWidget(network)
       break
     case NetworkGatewayType.MOON_PAY:
-      const widgetUrl = new MoonPay(moonPayConfig).widgetUrl(address, network)
+      const moonPay = new MoonPay(moonPayConfig)
+      const widgetUrl: string = yield call(moonPay.widgetUrl, address, network)
+      // TODO (sign url): what does happen if the request fail?
```
Comment (LautaroPetaccio) [inline src/modules/manaFiatGateway/sagas.ts:61]: Ideally, we should have an action to start a loading screen for the request and success and failure actions accordingly to notify the UI of what's going on.

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

File: src/modules/manaFiatGateway/sagas.ts [hunk 2]
```diff
@@ -44,23 +50,35 @@ export function createManaFiatGatewaysSaga(config: ManaFiatGatewaySagasConfig) {
...
+          address,
+          network
+        )
+        window.open(widgetUrl, '_blank', 'noopener,noreferrer')
+        break
+      default:
+        break
+    }
+
+    yield put(openManaFiatGatewaySuccess())
```
Comment (LautaroPetaccio) [inline src/modules/manaFiatGateway/sagas.ts:79]: Would you mind adding saga tests that exercise the put of the `openManaFiatGatewaySuccess` and `openManaFiatGatewayFailure` actions?

**Review by LautaroPetaccio — CHANGES_REQUESTED** [review]

File: src/modules/manaFiatGateway/reducer.spec.ts [hunk 3]
```diff
@@ -1,12 +1,72 @@
...
+      { loading: [], error: 'error' },
+      action
+    )
+
+    expect(state).toEqual({ loading: [action], error: null })
+  })
+})
+
+describe('when handling the open mana fiat gateway modal success', () => {
+  it('should update data and remove the request action from the loading state', () => {
```
Comment (LautaroPetaccio) [inline src/modules/manaFiatGateway/reducer.spec.ts:32]:
```suggestion
  it('should remove the request action from the loading state and remove the error', () => {
```

**Review by LautaroPetaccio — APPROVED** [review]

File: src/modules/manaFiatGateway/moonpay/MoonPay.spec.ts [hunk 4]
```diff
@@ -136,25 +134,15 @@ describe('when interacting with MoonPay', () => {
...
-  describe('when build the widget url', () => {
-    it('should return the entire url using the information from the config and the address of the connected user', () => {
-      const redirectUrl = `${window.location.origin}?network=${Network.ETHEREUM}&gateway=${NetworkGatewayType.MOON_PAY}`
-      return expect(moonPay.widgetUrl(mockAddress, Network.ETHEREUM)).toEqual(
-        `${mockConfig.widgetBaseUrl}?apiKey=${
-          mockConfig.apiKey
-        }&currencyCode=MANA&walletAddres=${mockAddress}&redirectURL=${encodeURIComponent(
-          redirectUrl
-        )}`
+  describe.only('when build the widget url', () => {
```
Comment (fzavalia) [inline src/modules/manaFiatGateway/moonpay/MoonPay.spec.ts:139]: Remove only

**Review by fzavalia — CHANGES_REQUESTED** [review]

File: src/modules/manaFiatGateway/sagas.ts [hunk 5]
```diff
@@ -44,23 +50,31 @@ export function createManaFiatGatewaysSaga(config: ManaFiatGatewaySagasConfig) {
...
-      window.open(widgetUrl, '_blank', 'noopener,noreferrer')
-      break
-    default:
-      break
+  try {
+    const address: string = yield select(getAddress)
+
+    switch (gateway) {
+      case NetworkGatewayType.TRANSAK:
+        const transak = new Transak(transakConfig, address, network)
```
Comment (juanmahidalgo) [inline src/modules/manaFiatGateway/sagas.ts:63]:
```suggestion
        const address: string = yield select(getAddress)
        const transak = new Transak(transakConfig, address, network)
```
I'd move the address selector to the `transak` case now that the moonpay one won't use it anymore, wdyt?

**Review by fzavalia — APPROVED** [review]

**Pull request merged**